### PR TITLE
Feat/a2 1191 rule 6 login email

### PR DESCRIPTION
**Pull request by topickering**
## Ticket Number

https://pins-ds.atlassian.net/browse/A2-1191
https://pins-ds.atlassian.net/browse/A2-1221
https://pins-ds.atlassian.net/browse/A2-1207
https://pins-ds.atlassian.net/browse/A2-1204
https://pins-ds.atlassian.net/browse/A2-1202
https://pins-ds.atlassian.net/browse/A2-1223
https://pins-ds.atlassian.net/browse/A2-1239

## Description of change

Adds the new log-in routes for Rule 6 Users

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- X My commit history in this PR is linear
- [X New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/controllers/rule-6/enter-code.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+ * @property {boolean} isGeneralLogin - defines if this enter code journey is for a general appeal log in, unrelated to an appeal
+ */
+
+/**
+ * Creates a one time password grant via the auth server
+ * @param {string} email
+ * @param {string} action
+ * @returns {Promise<void>}
+ */
+const createOTPGrant = async (email, action) => {
```
Comment (bryn-pins) [inline packages/forms-web-app/src/controllers/rule-6/enter-code.js:26]: lets reuse this function in auth-client perhaps

**Review by bryn-pins — COMMENTED** [review]
looks fairly clean
Only thing I think we're missing is a means to know if it's a rule 6 enabled user or not, I think currently this allows either an appellant or LPA user into the rule6 dashboard, which I _think_ we should stop?

probably a case of separate middleware
and a separate createRule6UserSession in user service, that checks they have a rule6 role on any appeal

**Comment by topickering** [comment]
Yes, I did wonder whether we want to check for R6 Users specifically via middleware - we've not moved on the the dashboard yet but that would only show appeals linked to that Rule6 user, but I suppose a benefit of the middleware might be that it wouldn't even allow them to see an empty dashboard.

File: packages/forms-web-app/src/controllers/rule-6/enter-code.js (hunk 1 above)
Comment (topickering) [reply packages/forms-web-app/src/controllers/rule-6/enter-code.js:26]: Good idea - there is a fair amount of common code which could be refactored here I think.

**Review by bryn-pins — APPROVED** [review]

**Pull request merged**